esp/platform/espprotocol.cpp

```diff
@@ -90,6 +91,29 @@ void CEspApplicationPort::appendBinding(CEspBindingEntry* entry, bool isdefault)
     }
 }
 
+void CEspApplicationPort::removeBinding(IEspRpcBinding* binding)
+{
+    WriteLockBlock wblock(rwLock);
+    for (int i = 0; i < bindingCount; i++)
+    {
+        if (!bindings[i])
+            continue;
+        IEspRpcBinding* currentBinding = bindings[i]->queryBinding();
+        if (currentBinding && currentBinding == binding)
+        {
+            Owned<CDelayedBindingRemover> remover = new CDelayedBindingRemover(bindings[i]);
```

Comment:
This feels iffy. You are starting a thread that will have no owner - will it leak? Will it release itself when it completes? Why does it need releasing asynchronously anyway?

Reply:
The run method of the thread links at the beginning and releases at the end. But you're right there might be a chance the object will be released before the link happens. The asynchronous release was put in place out of paranoia that the binding could be released while requests are still being processed. But since a holder for the binding was added in the request processing method, this shouldn't be a concern. I'll just remove the thread.

Reply:
You may want to arrange for the release to happen separately from the removal from the list - so that you don't hold onto the write lock any longer than you need to. But that does not require a separate thread.